### PR TITLE
vc: Don't adjust block index on error

### LINKED ISSUE #2244: Sandbox#getAndSetSandboxBlockIndex shouldn't alter BlockIndex in case of error
# Description of problem

Currently Sandbox#getAndSetSandboxBlockIndex increments BlockIndex up front.
Later on if store.Store() returns error, the index increment is left behind.

# Expected result

BlockIndex should not increment in case of error.

Similar issue applies to decrementSandboxBlockIndex where the symptom may be more severe - in worst case, the block index may become negative.

# Actual result

### PR DESCRIPTION
**Pull request by tedyu**
Fixed `Sandbox::getAndSetSandboxBlockIndex() so that
it does not update `BlockIndex` if an error occurs.

Fixes #2244

Signed-off-by: Ted Yu <yuzhihong@gmail.com>

### CODE REVIEW

**Comment by tedyu** [comment]
```
10:46:28 ERROR: kata-log-parser: no log records to process
10:46:28 POST BUILD TASK : FAILURE
```
I don't see any test failing.

@devimc @jcvenegas 
Can you trigger another test run ?

Thanks

**Comment by tedyu** [comment]
```
13:13:05 mv: cannot stat 'kata-qemu-static.tar.gz': No such file or directory
13:13:05 Failed at 46: sudo mv ${QEMU_TAR} ${KATA_TESTS_CACHEDIR}
13:13:05 Failed at 37: "${cidir}/install_qemu.sh"
```
I don't think the above was related to my change.

File: virtcontainers/sandbox.go [hunk 1]
```diff
@@ -1605,6 +1605,12 @@ func (s *Sandbox) resumeSetStates() error {
 // block device is assigned to a container in the sandbox.
 func (s *Sandbox) getAndSetSandboxBlockIndex() (int, error) {
 	currentIndex := s.state.BlockIndex
+	var err error
```
Comment (bergwolf) [inline virtcontainers/sandbox.go]: err is redefined and not set below.

**Comment by tedyu** [comment]
@bergwolf 
Your comment has been addressed.

**Comment by tedyu** [comment]
@jodh-intel @grahamwhaley 
Can you merge this - all tests passed.

**Comment by grahamwhaley** [comment]
@tedyu - just fyi, the Kata project runs a rolling rota of PR reviewers, which you can find on the wiki page at https://github.com/kata-containers/community/wiki/Review-Team-Rota.
Please allow reviewers some time (a few days) to look at which PRs are pending and review/merge them. If you get no response or updates for a few days, **then** feel free to ping either somebody relevant to the code (see https://github.com/kata-containers/community/wiki/Areas-of-interest for a list of who generally works on or is responsible for what), or one of the reviewers for that week.
 thanks.

**Pull request merged**